runtime/middleware/jwt.go

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2024 Hypermode Inc.
+ * Licensed under the terms of the Apache License, Version 2.0
+ * See the LICENSE file that accompanied this code for further details.
+ *
+ * SPDX-FileCopyrightText: 2024 Hypermode Inc. <hello@hypermode.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package middleware
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/json"
+	"encoding/pem"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/hypermodeinc/modus/runtime/config"
+	"github.com/hypermodeinc/modus/runtime/logger"
+	"github.com/hypermodeinc/modus/runtime/utils"
+
+	"github.com/golang-jwt/jwt/v5"
+)
+
+type jwtClaimsKey string
+
+const jwtClaims jwtClaimsKey = "jwt_claims"
+
+var authPublicKeys map[string]any
+
+func Init(ctx context.Context) {
+	publicKeysJson := os.Getenv("MODUS_PEMS")
+	if publicKeysJson == "" {
```

Comment:
We should log this scenario for the case when we specify the env key wrong by mistake